tests/test_reduction.py

```diff
@@ -226,6 +226,52 @@ def _test_area_integrated_average(solver1, solver2):
     solver1.setup.named_expressions.pop(key="test_expr_1")
 
 
+def _test_error_handling(solver):
+    with pytest.raises(RuntimeError) as msg:
+        reduction.area_average(
+            expr="AbsoluteVelocity",  # This is a wrong expression intentionally passed
+            locations=solver.setup.boundary_conditions.velocity_inlet,
+        )
+
+    assert msg.value.args[0] == "Unable to evaluate expression"
+
+
+def _test_force(solver):
+    solver.solution.initialization.hybrid_initialize()
+    solver.setup.named_expressions["test_expr_1"] = {}
+    solver.setup.named_expressions["test_expr_1"].definition = "Force(['wall'])"
+    expr_val_1 = solver.setup.named_expressions["test_expr_1"].get_value()
+
+    red_total_force = reduction.force(locations=[solver.setup.boundary_conditions.wall])
+    red_pressure_force = reduction.pressure_force(locations=["wall"], ctxt=solver)
+    red_viscous_force = reduction.viscous_force(
+        locations=[solver.setup.boundary_conditions.wall]
+    )
+
+    assert (red_total_force == expr_val_1).all()
+
+    assert (red_pressure_force + red_viscous_force == red_total_force).all()
+
+    solver.setup.named_expressions.pop(key="test_expr_1")
+
+
+def _test_moment(solver):
+    solver.solution.initialization.hybrid_initialize()
+    solver.setup.named_expressions["test_expr_1"] = {}
+    solver.setup.named_expressions[
+        "test_expr_1"
+    ].definition = "Moment(Force(['wall']),['wall'])"
+    expr_val_1 = solver.setup.named_expressions["test_expr_1"].get_value()
+
+    red_moment = reduction.moment(
+        expr="Force(['wall'])", locations=[solver.setup.boundary_conditions.wall]
```

Comment:
@seanpearsonuk, for now the moment can be calculated by passing the expression like: "Force(['wall'])", as shown in the test case. However, we can also do something to process out the reduction string from reduction.force(<locations>), and then pass that sting to 'expr' of reduction.moment(). For now the later has not been applied for keeping the behavior analogous to others where 'expr' was a string. Please let me know what you think and whether the later should be applied (as it looks interesting :)).
Thank you.

Reply:
@prmukherj above usage looks good. expr argument can in general be any expression accepted as valid by the backend implementation so expr can be an API object too. I don't know if "process out the reduction string from reduction.force()" works or not.